fog/sample-paykit/src/cached_tx_data/mod.rs

```diff
@@ -475,31 +481,33 @@ impl CachedTxData {
         }
 
         self.missed_block_ranges.extend(new_missed_block_ranges);
-        let fog_common_block_ranges: Vec<fog_common::BlockRange> = self
-            .missed_block_ranges
-            .iter()
-            .map(fog_common::BlockRange::from)
-            .collect::<Vec<_>>();
-        match fog_block_client.get_missed_block_ranges(fog_common_block_ranges) {
-            Ok(block_response) => {
-                let tx_out_records_from_missed_blocks: Vec<TxOutRecord> =
-                    self.create_tx_out_records(&block_response);
-                txo_records.extend(tx_out_records_from_missed_blocks);
-                let updated_missed_block_ranges =
-                    CachedTxData::calculate_updated_missed_block_ranges(
-                        &self.missed_block_ranges,
-                        &block_response.blocks.into_vec(),
+        if !self.missed_block_ranges.is_empty() {
+            let fog_common_block_ranges: Vec<fog_common::BlockRange> = self
+                .missed_block_ranges
+                .iter()
+                .map(fog_common::BlockRange::from)
+                .collect::<Vec<_>>();
+            match fog_block_client.get_missed_block_ranges(fog_common_block_ranges) {
```

Comment:
:question: This is mainly a thought exercise, I think both ways are reasonable solutions, just want to be sure this implementation isn't only the result of the easiest thing to do at the time.

Was there any consideration to putting logic in `get_missed_block_ranges()` that sees if the ranges is an empty vec and returns an empty response?

From the consumer of the API it behaves the same, other than it's a bit faster.
A downside is if the response is used for more than just the missed ranges.

Reply:
I considered that, but ended up not touching the actual call, mainly because we haven't done that in other places (which may or may not be a good reason), as well as callers might not expect that (e.g. if this is exercised in integration tests and we wanted to test how the server responds to an empty request). In reality either way is probably acceptable...